libc/include/assert.h.def

```diff
@@ -6,9 +6,16 @@
 //
 //===----------------------------------------------------------------------===//
 
+#ifndef LLVM_LIBC_ASSERT_H
+#define LLVM_LIBC_ASSERT_H
```

Comment:
assert.h is funny; it's the one header that should NEVER use header guards, because it's possible to have consumers change whether `NDEBUG` is defined, then re-include this header.

Adding header guards to assert.h will break that quite unfrequently used IME feature.

Reply:
Ah, didn't know that, pretty interesting :o